aspen/request_processor/__init__.py

```diff
@@ -23,7 +23,7 @@
     # 'name':               (default,               from_unicode)
 KNOBS = \
     { 'changes_reload':     (False,                 parse.yes_no)
-    , 'charset_dynamic':    ('UTF-8',               parse.charset)
+    , 'charset_dynamic':    ('UTF-8',               parse.codec)
     , 'charset_static':     (None,                  parse.charset)
```

Comment:
`charset_dynamic` needs to be a codec that python supports because it's used to encode `unicode` outputs. On the other hand `charset_static` is purely declarative so we don't need to restrict it to the set of codecs that python implements.